### ISSUE TITLE
Cross-compilation on Mac OS for building Docker container

### ISSUE THREAD

**Issue by samgiles** [fxbox/registration_server#27]
Instead of building inside the docker container, and then deploying - we should cross compile (or delete intermediate layers in the docker image?) so that the final image is minimal.  Maybe switch from debian to the alpine base container.

**Comment by michielbdejong** [comment]
Right, we could publish the built executable on some URL, and then just write a Dockerfile that `wget`s it.

**Comment by michielbdejong** [comment]
One advantage of building inside the Dockerfile is that this way the repo is self-documented by its Dockerfile: you only need Docker to be able to build. For instance: the Dockerfile now explicitly picks Rust nightly-2016-03-07; if you publish a built executable, then this information about the Rust version needs to be human-readable somewhere in the readme.

IMHO the ability to easily build the registration server is more important than disk space used due to image size, but I also understand if you think otherwise.
